deepdrivemd/sim/openmm_stream/openmm_reporter.py

```diff
@@ -64,11 +64,14 @@ def report(self, simulation, state):
         contact_map = t2Dto1D(contact_map)
         contact_map = np.packbits(contact_map)
 
-        mda_u = MDAnalysis.Universe(str(self.cfg.reference_pdb_file))
-        reference_positions = mda_u.select_atoms(
-            self.cfg.mda_selection
-        ).positions.copy()
-        rmsd = rms.rmsd(positions_ca, reference_positions, superposition=True)
+        if self.cfg.compute_rmsd:
+            mda_u = MDAnalysis.Universe(str(self.cfg.reference_pdb_file))
+            reference_positions = mda_u.select_atoms(
+                self.cfg.mda_selection
+            ).positions.copy()
+            rmsd = rms.rmsd(positions_ca, reference_positions, superposition=True)
+        else:
+            rmsd = -1.0
```

Comment:
It's better now that the computation is optional. We can further improve it by optionally sending the data via ADIOS as well. In general we may have a multiple order parameters like (rmsd, franction of native contacts, center of mass distances, radius of gyration, etc). So we really need the ability to optionally write data to the ADIOS stream. We shouldn't be writing fake data (-1.0) to the stream, since then we would have to do that for each different type of order parameter supported.

Reply:
I think the fix is
```
if self.cfg.compute_rmsd:
    mda_u = MDAnalysis.Universe(str(self.cfg.reference_pdb_file))
    reference_positions = mda_u.select_atoms(
        self.cfg.mda_selection
    ).positions.copy()
    rmsd = rms.rmsd(positions_ca, reference_positions, superposition=True)
    rmsd = np.array([rmsd], dtype=np.float32)

step = np.array([step], dtype=np.int32)

output = {
    "md5": md5,
    "step": step,
    "positions": positions,
    "velocities": velocities,
    "contact_map": contact_map,
}

if self.cfg.compute_rmsd:
    output["rmsd"] = rmsd

self.write_adios_step(output)
self.step += 1
```

But perhaps other areas of the code need to be updated as well.

Reply:
Done: rmsd is removed from ADIOS streams when compute_rmsd=False. It still goes into OutlierDB not to break its structure but it is a tiny amount of data: number of entries is equal to the number of parallel simulations.